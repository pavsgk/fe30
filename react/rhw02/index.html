<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>

<script>
  const sendReq = async () => {
    const result = await fetch('https://script.google.com/macros/s/AKfycbzEYFV5JBtD5d2D2swAs6i9n0SeqMxAgZRD-tJG6F3ONbSgF38kL4qeuJ-U-FMDebvxew/exec?___testS1hop1').then(res => res.json());
    console.log(result);
    return result;
  }

  sendReq();



</script>  
</body>
</html>