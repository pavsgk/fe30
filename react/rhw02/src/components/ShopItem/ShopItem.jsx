import React from 'react';
import styles from './ShopItem.module.scss';
import { ReactComponent as FavSvgTrue } from './add-fav-true.svg';
import { ReactComponent as FavSvgFalse } from './add-fav-false.svg';


function ShopItem(props) {
  const {name, price, image, sku, color, id, isFav, backgroundColor, toggleFav, addCart, showModal} = props;

  return (
    <div className={styles.ShopItem} style={{backgroundColor}}>
      <img src={image}></img>
      <div onClick={() => {
        toggleFav(id);
      }}>
        {isFav ? <FavSvgTrue /> : <FavSvgFalse />}
      </div>
      <div>
        <h3>{name}</h3>
        <ul>
          <li>Color: {color}</li>
          <li>SKU: {sku}</li>
        </ul>
        <span>{price} â‚´</span>
      </div>
      <button onClick={() => 
        showModal(() => addCart(id), 'Add to cart?', 'Are you sure want to add this to cart?')
      }>Add to cart</button>
    </div>
  );
}

export default ShopItem;