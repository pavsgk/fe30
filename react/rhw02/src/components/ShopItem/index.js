import React, { useState, useEffect } from 'react';
import randomHexColor from '../../utils/randomHexColor.js';
import styles from './ShopItem.module.scss';
import { ReactComponent as FavSvgTrue } from './add-fav-true.svg';
import { ReactComponent as FavSvgFalse } from './add-fav-false.svg';


function ShopItem(props) {
  const {name, price, image, sku, color, id, isFav, backgroundColor, toggleFav, addCart} = props;

  return (
    <div className={styles.ShopItem} style={{backgroundColor}}>
      <img src={image}></img>
      <div onClick={() => {
        toggleFav(id);
      }}>
        {isFav ? <FavSvgTrue /> : <FavSvgFalse />}
      </div>
      <div>
        <h3>{name}</h3>
        <ul>
          <li>Color: {color}</li>
          <li>SKU: {sku}</li>
        </ul>
        <span>{price} â‚´</span>
      </div>
      <button onClick={() => addCart(id)}>Add to cart</button>
    </div>
  );
}

export default ShopItem;