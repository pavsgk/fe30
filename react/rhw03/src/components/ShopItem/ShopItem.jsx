import React from 'react';
import styles from './ShopItem.module.scss';
import { ReactComponent as FavSvgTrue } from './add-fav-true.svg';
import { ReactComponent as FavSvgFalse } from './add-fav-false.svg';
import PropTypes from 'prop-types';


function ShopItem(props) {
  const {name, price, image, sku, color, id, isFav, backgroundColor, toggleFav, addCart, removeCart, showModal} = props;
  return (
    <div className={styles.ShopItem} style={{backgroundColor}}>
      <img src={image} alt={name}></img>
      <div onClick={() => {
        toggleFav(id);
      }}>
        {isFav ? <FavSvgTrue /> : <FavSvgFalse />}
      </div>
      <div>
        <h3>{name}</h3>
        <ul>
          <li>Color: {color}</li>
          <li>SKU: {sku}</li>
        </ul>
        <span>{price} â‚´</span>
      </div>
      {addCart && <button onClick={() => 
        showModal(() => addCart(id), 'Add to cart?', 'Are you sure want to add this to cart?')
      }>Add to cart</button>}
      {removeCart && <button onClick={() => 
        showModal(() => removeCart(id), 'Remove from cart?', 'Are you sure want to remove this from cart?')
      }>Remove from cart</button>}
    </div>
  );
}

ShopItem.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  image: PropTypes.string.isRequired,
  sku: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,

  toggleFav: PropTypes.func.isRequired,
  addCart: PropTypes.func,
  removeCart: PropTypes.func,
  showModal: PropTypes.func.isRequired,

  backgroundColor: PropTypes.string,
  isFav: PropTypes.bool,
}

ShopItem.defaultProps = {
  removeCart: undefined,
  addCart: undefined,
  backgroundColor: 'lightgrey',
  isFav: false
}

export default ShopItem;