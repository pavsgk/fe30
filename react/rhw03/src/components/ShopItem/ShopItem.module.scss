.ShopItem {
  border-radius: 15px;
  padding: 5px;
  position: relative;

  img {
    max-height: 200px;
    object-fit: cover;
    border: 1px solid gray;
    border-radius: 15px;
  }

  svg {
    position: absolute;
    right: 15px;
    top: 15px;
    cursor: pointer;
    
    path {
      fill: firebrick;
    }
  }
}