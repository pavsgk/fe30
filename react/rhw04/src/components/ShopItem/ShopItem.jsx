import React from 'react';
import styles from './ShopItem.module.scss';
import { ReactComponent as FavSvgTrue } from './add-fav-true.svg';
import { ReactComponent as FavSvgFalse } from './add-fav-false.svg';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { showModal, toggleFavGoods, addToCart, deleteFromCart } from '../../store/actionCreators';

function ShopItem(props) {
  const dispatch = useDispatch();
  const {shop, cart, name, price, image, sku, color, id, backgroundColor, index} = props;
  const isFav = useSelector(store => store.goods.items[index].isFav);

  return (
    <div className={styles.ShopItem} style={{backgroundColor}}>
      <img src={image} alt={name}></img>
      <div onClick={() => {
        dispatch(toggleFavGoods(id));
      }}>
        {isFav ? <FavSvgTrue /> : <FavSvgFalse />}
      </div>
      <div>
        <h3>{name}</h3>
        <ul>
          <li>Color: {color}</li>
          <li>SKU: {sku}</li>
        </ul>
        <span>{price} â‚´</span>
      </div>
      {shop && <button onClick={() => {
        dispatch(showModal({title: 'Add to cart?', text: 'Are you sure want to add this to cart?', actionFn: () => {
          dispatch(addToCart(id));
        }}));
      }}>Add to cart</button>}
      {cart && <button onClick={() => {
        dispatch(showModal({title: 'Remove from cart?', text: `Are you sure want to remove this sku (${sku}) from cart?`, actionFn: () => {
          dispatch(deleteFromCart(id));
        }}));
      }}>Remove from cart</button>}
    </div>
  );
}

ShopItem.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  image: PropTypes.string.isRequired,
  sku: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,

  backgroundColor: PropTypes.string,
  isFav: PropTypes.bool,
}

ShopItem.defaultProps = {
  removeCart: undefined,
  addCart: undefined,
  backgroundColor: 'lightgrey',
  isFav: false
}

export default ShopItem;