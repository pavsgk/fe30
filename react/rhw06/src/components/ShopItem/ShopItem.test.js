import ShopItem from "./ShopItem";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

const testStore = {
  goods: {
    items: [
      {
        id: 0,
        name :"Birhday",
        price :"290.95",
        image :"https://media.istockphoto.com/photos/birthday-cake-picture-id812908632?k=20&m=812908632&s=612x612&w=0&h=r_HtCSEJe4vz2c-RX-rPChFY9l50nsl9PXdGRoqkJmQ=",
        sku :"bc101",
        color :"purple",
        isFav : true,
      }]
  },
  favRerender: false,
  getState() {
    return this;
  },
  subscribe() {},
  dispatch() {},
};

describe("Snapshot test", () => {
  test("Should component render", () => {
    
    const props = testStore.goods.items[0];

    const { asFragment } = render(
      <Provider store={testStore}>
        <ShopItem index={0} {...props}/>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
