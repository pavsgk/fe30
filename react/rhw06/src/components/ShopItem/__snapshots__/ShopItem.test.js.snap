// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Should component render 1`] = `
<DocumentFragment>
  <div
    class="ShopItem"
    style="background-color: lightgrey;"
  >
    <img
      alt="Birhday"
      src="https://media.istockphoto.com/photos/birthday-cake-picture-id812908632?k=20&m=812908632&s=612x612&w=0&h=r_HtCSEJe4vz2c-RX-rPChFY9l50nsl9PXdGRoqkJmQ="
    />
    <div>
      <svg>
        add-fav-true.svg
      </svg>
    </div>
    <div>
      <h3>
        Birhday
      </h3>
      <ul>
        <li>
          Color: purple
        </li>
        <li>
          SKU: bc101
        </li>
      </ul>
      <span>
        290.95 ₴
      </span>
      <p>
         
      </p>
    </div>
  </div>
</DocumentFragment>
`;
